### DISCUSSION TITLE
Limiting registration and access to projects

### DISCUSSION THREAD

**Discussion by impact-computing** [NationalSecurityAgency/skills-service#1016] [Q&A] [answered]
Hello all,

Firstly thank you for putting this together. I've been looking at something like this for a while and there are so many great features.

The way I am hoping to utilise this is by having all users log in to the Dashboard. I've read through the documentation but I can't seem to find a way to turn on registration approval or anything like that. It seems if skilltree is publicly accessible there is no way to prevent registrations utilising the standard password auth?

As a side note, is it possible to limit projects to certain users or is it the case that at any point any user can start earning towards a project.

**Comment by sudo-may** [comment]
@impact-computing Thank you for the kind words! 

Is your plan to enable [Progress & Ranking](https://skilltreeplatform.dev/dashboard/user-guide/progress-and-ranking.html) views and then have your users (after logging in) utilize those views to discover what projects, subjects, skills, etc... can be achieved? In this case they would also use these views to see their progress and ranking. Just wanted to make sure we understood your plan correctly. 

There is currently no registration approval supported. It's been discussed several times internally so I went ahead and created tickets to support these features: https://github.com/NationalSecurityAgency/skills-service/issues/1017, https://github.com/NationalSecurityAgency/skills-service/issues/1018. Note that a user with a Dashboard account can only edit, via the Project Administration section of the dashboard, projects that they have created or projects for which they have been added as a Project Administrator.

There is also no way to add project-based access restriction for [Progress and Ranking](https://skilltreeplatform.dev/dashboard/user-guide/progress-and-ranking.html) viewing or limit skill reporting to only certain users. If users have a dashboard account they are able to view and report (https://github.com/NationalSecurityAgency/skills-service/issues/1018 addresses that). However, only projects with the [Discoverable](https://skilltreeplatform.dev/dashboard/user-guide/projects.html#settings) setting enabled are presented in the available [My Projects](https://skilltreeplatform.dev/dashboard/user-guide/progress-and-ranking.html#manage-my-projects-page) catalog. So if the [Discoverable](https://skilltreeplatform.dev/dashboard/user-guide/projects.html#settings) setting is not enabled there is no way for users to locate that project (unless they know the project id). In this case the project can only be accessed through a direct url (ex. https:<skill tree dashboard install>/progress-and-rankings/projects/<project id>) which can be emailed to only select users. This url can be obtained by the project administrator by clicking the Preview button on the project's admin page. Not sure if that would meet your needs but wanted to present that option.  

Hope this helps!

**Comment by impact-computing** [comment]
Hi - yes that was my plan. I basically want to use it as a self contained app for all users. It appears in 1.8.1 this now needs to be added in to the configuration settings to get it turned on? When new users sign up in 1.8.1 they just get an empty project admin  page (not sure why it's defaulting to the Admin page/header?). I accidentally deployed 1.4.something earlier and this worked much better from an experience point of view. It's not 100% clear how to make that configuration change to get it back to that view. 

To give you real life examples to help illustrate usage, we're an IT company who has separate levels of engineers. There are different skills for say 1st Line Support, 2nd Line Support, etc. We have defined requirements (including qualifications needed) for people to pass from one to the next. I wouldn't want a 1st Line Support engineer going through the 2nd Line engineer stuff. I ideally want one way for them to login (preferably extended to Azure AD SSO) that they can then consume the skills/projects from.

Hope that described my use case. I know some of it is not relevant but just wanted to put it across in-case it helps.

**Reply by sudo-may** [reply]
@impact-computing Thank you for the detailed explanation. I think in the 1.7 release Progress and Ranking views were disabled by default as they are not needed on an install where embedded components (via SkillTree client libs) are integrated into an application. You can set the following properties to enable and to default to the P&R view: 

``` properties
# enable Progress and Ranking Views
skills.config.ui.rankingAndProgressViewsEnabled=true

# optionally change default for the landing page (admin is the default)
skills.config.ui.defaultLandingPage=progress
```
Docs links: https://skilltreeplatform.dev/dashboard/install-guide/config.html#progress-and-ranking-views

**Comment by sudo-may** [comment, marked as answer]
Hello @impact-computing, 

[skills-service:2.0](https://skilltreeplatform.dev/release-notes/skills-service.html#_2-0-0) introduced the [Private Invite Only Project](https://skilltreeplatform.dev/dashboard/user-guide/projects.html#setting-visibility) feature - Progress and Ranking pages can only be accessed by users who have been invited to join the project and who have accepted the invite, any other user attempting to access the project will receive an Access Denied error.
